Tests/MongoSwiftTests/AsyncAwaitTests.swift

```diff
@@ -114,4 +115,161 @@ final class AsyncAwaitTests: MongoSwiftTestCase {
     }
 }
 
+final class MongoCursorAsyncAwaitTests: MongoSwiftTestCase {
+    func testAsyncSequenceConformance() throws {
+        testAsync {
+            try await self.withTestNamespace { _, _, coll in
+                let docsToInsert: [BSONDocument] = [["_id": 1], ["_id": 2], ["_id": 3]]
+                try await coll.insertMany(docsToInsert)
+
+                // iterating via a for loop
+                var foundDocs = [BSONDocument]()
+                let sortOpts = FindOptions(sort: ["_id": 1])
+                for try await doc in try await coll.find(options: sortOpts) {
+                    foundDocs.append(doc)
+                }
+                expect(foundDocs).to(equal(docsToInsert))
+
+                // manual iteration with next()
+                let cursor = try await coll.find(options: sortOpts)
+                let first = try await cursor.next()
+                expect(first).to(equal(docsToInsert[0]))
+                let second = try await cursor.next()
+                expect(second).to(equal(docsToInsert[1]))
+                let third = try await cursor.next()
+                expect(third).to(equal(docsToInsert[2]))
+                let fourth = try await cursor.next()
+                expect(fourth).to(beNil())
+                let isAlive = try await cursor.isAlive()
+                expect(isAlive).to(beFalse())
+            }
+        }
+    }
+
+    // Test that a tailable cursor that is continually polling the server can be killed by cancelling the parent Task.
+    func testTailableCursorHandlesTaskCancellation() throws {
```

Comment:
I had mentioned on Slack creating the cursor outside of the child task and then asserting on `isAlive()`, however the thread sanitizer did not like that. I think this still covers what we are trying to test though, to your point @patrickfreed about assuming tailable cursors work right.